### ISSUE TITLE
RC 32 WHERE Time Failure

### ISSUE THREAD

**Issue by Jhors2** [influxdb/influxdb#2836]
I am currently running a query as follows:
```SELECT sum(value) FROM cpu_load WHERE server =~ /^http/ AND time > now() - 1m GROUP BY time(60s)```

Whenever I attempt a query using the now() or any other time function it appears to be broken.  I am still trying to troubleshoot to verify it is only the now() function that appears to be broken.

**Comment by beckettsean** [comment]
@Jhors2 Can you clarify what you mean by "appears to be broken"? Is there an error output or a stack trace, or is it just an empty return?

**Comment by Jhors2** [comment]
@beckettsean 
Sure, I thought at first it was an empty return, however there is a stack trace see here:
```goroutine 52 [running]:
github.com/influxdb/influxdb/influxql.MapSum(0x7f31b8647800, 0xc208b6de40, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/influxql/functions.go:424 +0xac
github.com/influxdb/influxdb/tsdb.(*LocalMapper).NextInterval(0xc208b6de40, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tsdb/tx.go:378 +0xd3
github.com/influxdb/influxdb/influxql.(*MapReduceJob).processAggregate(0xc20a0a6150, 0xc20a168600, 0xb6cc48, 0xc209b68000, 0x5a0, 0x5a0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/influxql/engine.go:778 +0x264
github.com/influxdb/influxdb/influxql.(*MapReduceJob).Execute(0xc20a0a6150, 0xc209fce540, 0x6800)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/influxql/engine.go:170 +0x8ca
github.com/influxdb/influxdb/influxql.(*Executor).execute(0xc209948f40, 0xc209fce540)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/influxql/engine.go:926 +0xba
created by github.com/influxdb/influxdb/influxql.(*Executor).Execute
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/influxql/engine.go:905 +0x5a

goroutine 1 [chan receive (nil chan)]:
main.(*Main).Run(0xc2080d9f60, 0xc20800a010, 0x4, 0x4, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/main.go:60 +0x3d6
main.main()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/main.go:24 +0x1d0

goroutine 6 [syscall]:
os/signal.loop()
	/root/.gvm/gos/go1.4.2/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/root/.gvm/gos/go1.4.2/src/os/signal/signal_unix.go:27 +0x35

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/root/.gvm/gos/go1.4.2/src/runtime/asm_amd64.s:2232 +0x1

goroutine 8 [IO wait]:
net.(*pollDesc).Wait(0xc2080104c0, 0x72, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc2080104c0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc208010460, 0x0, 0x7f31bc49bde0, 0xc208100000)
	/root/.gvm/gos/go1.4.2/src/net/fd_unix.go:419 +0x40b
net.(*TCPListener).AcceptTCP(0xc20802c7f0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/tcpsock_posix.go:234 +0x4e
net.(*TCPListener).Accept(0xc20802c7f0, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/tcpsock_posix.go:244 +0x4c
github.com/influxdb/influxdb/tcp.(*Mux).Serve(0xc2080b9710, 0x7f31bc4a6068, 0xc20802c7f0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tcp/mux.go:45 +0x57
created by github.com/influxdb/influxdb/cmd/influxd/run.func·003
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:210 +0x76b

goroutine 9 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc20802c7f8, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tcp/mux.go:117 +0x80
github.com/influxdb/influxdb/meta.(*raftLayer).Accept(0xc2080b9740, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/meta/store.go:1539 +0x64
github.com/hashicorp/raft.(*NetworkTransport).listen(0xc208010540)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/net_transport.go:346 +0x50
created by github.com/hashicorp/raft.NewNetworkTransport
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/net_transport.go:138 +0x2c4

goroutine 10 [select]:
github.com/hashicorp/raft.(*Raft).leaderLoop(0xc2080b41c0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:807 +0xc7c
github.com/hashicorp/raft.(*Raft).runLeader(0xc2080b41c0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:779 +0x74e
github.com/hashicorp/raft.(*Raft).run(0xc2080b41c0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:576 +0xc9
github.com/hashicorp/raft.*Raft.(github.com/hashicorp/raft.run)·fm()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:241 +0x27
github.com/hashicorp/raft.func·008()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/state.go:152 +0x51
created by github.com/hashicorp/raft.(*raftState).goFunc
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/state.go:153 +0xe3

goroutine 11 [select]:
github.com/hashicorp/raft.(*Raft).runFSM(0xc2080b41c0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:490 +0xf74
github.com/hashicorp/raft.*Raft.(github.com/hashicorp/raft.runFSM)·fm()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:242 +0x27
github.com/hashicorp/raft.func·008()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/state.go:152 +0x51
created by github.com/hashicorp/raft.(*raftState).goFunc
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/state.go:153 +0xe3

goroutine 12 [select]:
github.com/hashicorp/raft.(*Raft).runSnapshots(0xc2080b41c0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:1588 +0x466
github.com/hashicorp/raft.*Raft.(github.com/hashicorp/raft.runSnapshots)·fm()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/raft.go:243 +0x27
github.com/hashicorp/raft.func·008()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/state.go:152 +0x51
created by github.com/hashicorp/raft.(*raftState).goFunc
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/hashicorp/raft/state.go:153 +0xe3

goroutine 13 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc20802c800, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tcp/mux.go:117 +0x80
github.com/influxdb/influxdb/meta.(*Store).serveExecListener(0xc20809a140)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/meta/store.go:416 +0x8f
created by github.com/influxdb/influxdb/meta.(*Store).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/meta/store.go:175 +0x196

goroutine 14 [select]:
github.com/influxdb/influxdb/cmd/influxd/run.(*Server).monitorErrorChan(0xc208098120, 0xc20802e420)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:271 +0x149
created by github.com/influxdb/influxdb/cmd/influxd/run.func·003
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:216 +0x8dc

goroutine 15 [select]:
github.com/influxdb/influxdb/services/hh.(*Service).retryWrites(0xc20803b040)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/hh/service.go:92 +0x2e7
created by github.com/influxdb/influxdb/services/hh.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/hh/service.go:61 +0xef

goroutine 16 [select]:
github.com/influxdb/influxdb/services/hh.(*Service).expireWrites(0xc20803b040)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/hh/service.go:110 +0x2a4
created by github.com/influxdb/influxdb/services/hh.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/hh/service.go:62 +0x109

goroutine 34 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc20802c808, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tcp/mux.go:117 +0x80
github.com/influxdb/influxdb/cluster.(*Service).serve(0xc208010310)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cluster/service.go:79 +0x102
created by github.com/influxdb/influxdb/cluster.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cluster/service.go:56 +0x67

goroutine 35 [IO wait]:
net.(*pollDesc).Wait(0xc20a1d91e0, 0x72, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc20a1d91e0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc20a1d9180, 0x0, 0x7f31bc49bde0, 0xc20a158070)
	/root/.gvm/gos/go1.4.2/src/net/fd_unix.go:419 +0x40b
net.(*TCPListener).AcceptTCP(0xc209b52ed0, 0xc20805d880, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/tcpsock_posix.go:234 +0x4e
net.(*TCPListener).Accept(0xc209b52ed0, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/tcpsock_posix.go:244 +0x4c
net/http.(*Server).Serve(0xc209ff02a0, 0x7f31bc4a6068, 0xc209b52ed0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1728 +0x92
net/http.Serve(0x7f31bc4a6068, 0xc209b52ed0, 0x7f31b8646988, 0xc20a158050, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1606 +0xa1
github.com/influxdb/influxdb/services/admin.(*Service).serve(0xc2080b9500)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/admin/service.go:71 +0x147
created by github.com/influxdb/influxdb/services/admin.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/admin/service.go:39 +0xd9

goroutine 36 [select]:
github.com/influxdb/influxdb/services/continuous_querier.(*Service).backgroundLoop(0xc208010380)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/continuous_querier/service.go:146 +0x1fc
created by github.com/influxdb/influxdb/services/continuous_querier.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/continuous_querier/service.go:87 +0x1ed

goroutine 37 [IO wait]:
net.(*pollDesc).Wait(0xc20a1d92c0, 0x72, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc20a1d92c0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc20a1d9260, 0x0, 0x7f31bc49bde0, 0xc20a13eaa8)
	/root/.gvm/gos/go1.4.2/src/net/fd_unix.go:419 +0x40b
net.(*TCPListener).AcceptTCP(0xc209b52ed8, 0xc208113dc8, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/tcpsock_posix.go:234 +0x4e
net.(*TCPListener).Accept(0xc209b52ed8, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/tcpsock_posix.go:244 +0x4c
net/http.(*Server).Serve(0xc20a02cf00, 0x7f31bc4a6068, 0xc209b52ed8, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1728 +0x92
net/http.Serve(0x7f31bc4a6068, 0xc209b52ed8, 0x7f31b8646938, 0xc2080103f0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1606 +0xa1
github.com/influxdb/influxdb/services/httpd.(*Service).serve(0xc20808d8c0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/service.go:83 +0x79
created by github.com/influxdb/influxdb/services/httpd.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/service.go:51 +0x283

goroutine 38 [select]:
github.com/influxdb/influxdb/services/retention.(*Service).deleteShardGroups(0xc20802e7e0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/retention/service.go:64 +0x433
created by github.com/influxdb/influxdb/services/retention.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/retention/service.go:46 +0x67

goroutine 39 [select]:
github.com/influxdb/influxdb/services/retention.(*Service).deleteShards(0xc20802e7e0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/retention/service.go:98 +0x821
created by github.com/influxdb/influxdb/services/retention.(*Service).Open
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/retention/service.go:47 +0x81

goroutine 40 [select]:
github.com/influxdb/influxdb/cmd/influxd/run.(*Command).monitorServerErrors(0xc20808da00)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/run/command.go:110 +0x2d9
created by github.com/influxdb/influxdb/cmd/influxd/run.(*Command).Run
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/cmd/influxd/run/command.go:94 +0x87a

goroutine 50 [chan receive]:
github.com/influxdb/influxdb/services/httpd.(*Handler).serveQuery(0xc2080103f0, 0x7f31b8646f10, 0xc20a1174e0, 0xc209a56d00, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:241 +0x7c4
github.com/influxdb/influxdb/services/httpd.*Handler.(github.com/influxdb/influxdb/services/httpd.serveQuery)·fm(0x7f31b8646f10, 0xc20a1174e0, 0xc209a56d00, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:90 +0x4f
github.com/influxdb/influxdb/services/httpd.func·001(0x7f31b8646f10, 0xc20a1174e0, 0xc209a56d00)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:589 +0x78
net/http.HandlerFunc.ServeHTTP(0xc2080ee4e0, 0x7f31b8646f10, 0xc20a1174e0, 0xc209a56d00)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1265 +0x41
github.com/influxdb/influxdb/services/httpd.func·002(0x7f31b8646e38, 0xc20a1173e0, 0xc209a56d00)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:647 +0x288
net/http.HandlerFunc.ServeHTTP(0xc2080ddda0, 0x7f31b8646e38, 0xc20a1173e0, 0xc209a56d00)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1265 +0x41
github.com/influxdb/influxdb/services/httpd.func·003(0x7f31b8646e38, 0xc20a1173e0, 0xc209a56d00)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:656 +0xb0
net/http.HandlerFunc.ServeHTTP(0xc2080ee500, 0x7f31b8646e38, 0xc20a1173e0, 0xc209a56d00)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1265 +0x41
github.com/influxdb/influxdb/services/httpd.func·004(0x7f31b8646e38, 0xc20a1173e0, 0xc209a56d00)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:689 +0x31a
net/http.HandlerFunc.ServeHTTP(0xc2080ddde0, 0x7f31b8646e38, 0xc20a1173e0, 0xc209a56d00)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1265 +0x41
github.com/influxdb/influxdb/services/httpd.func·005(0x7f31b8646e38, 0xc20a1173e0, 0xc209a56d00)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:699 +0x179
net/http.HandlerFunc.ServeHTTP(0xc2080dde00, 0x7f31b8646e38, 0xc20a1173e0, 0xc209a56d00)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1265 +0x41
github.com/influxdb/influxdb/services/httpd.func·006(0x7f31b8646e38, 0xc20a1173c0, 0xc209a56d00)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:707 +0x123
net/http.HandlerFunc.ServeHTTP(0xc2080ee520, 0x7f31b8646e38, 0xc20a1173c0, 0xc209a56d00)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1265 +0x41
github.com/influxdb/influxdb/services/httpd.func·007(0x7f31b8646e00, 0xc209ef0140, 0xc209a56d00)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:717 +0x12c
net/http.HandlerFunc.ServeHTTP(0xc2080ee540, 0x7f31b8646e00, 0xc209ef0140, 0xc209a56d00)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1265 +0x41
github.com/bmizerany/pat.(*PatternServeMux).ServeHTTP(0xc20802c740, 0x7f31b8646e00, 0xc209ef0140, 0xc209a56d00)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/bmizerany/pat/mux.go:109 +0x21c
github.com/influxdb/influxdb/services/httpd.(*Handler).ServeHTTP(0xc2080103f0, 0x7f31b8646e00, 0xc209ef0140, 0xc209a56d00)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/services/httpd/handler.go:166 +0x256
net/http.serverHandler.ServeHTTP(0xc20a02cf00, 0x7f31b8646e00, 0xc209ef0140, 0xc209a56d00)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1703 +0x19a
net/http.(*conn).serve(0xc209ef0000)
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1204 +0xb57
created by net/http.(*Server).Serve
	/root/.gvm/gos/go1.4.2/src/net/http/server.go:1751 +0x35e

goroutine 51 [chan receive]:
github.com/influxdb/influxdb/tsdb.(*QueryExecutor).executeSelectStatement(0xc20808d840, 0x0, 0xc209c12000, 0xc209c9e1e2, 0x7, 0xc209fce300, 0x2710, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tsdb/query_executor.go:236 +0x1f1
github.com/influxdb/influxdb/tsdb.func·004()
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tsdb/query_executor.go:157 +0x727
created by github.com/influxdb/influxdb/tsdb.(*QueryExecutor).ExecuteQuery
	/root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tsdb/query_executor.go:211 +0x22b```

**Comment by dgnorton** [comment]
@Jhors2 can you reproduce this on `master`?  I haven't been able to reproduce it.

**Comment by dgnorton** [comment]
Okay, I can reproduce it with `SELECT sum(some_string_field)...`

**Comment by Jhors2** [comment]
It is an aggregation function issue, yes.  I was mistaken, I had thought initially aggregation had worked when not specifying time fields.  #2845 would be the correct reference here, or this can have the issue name changed.

**Comment by beckettsean** [comment]
Believed fixed by #2859
